docs/source/design/dataset_argument.rst

```diff
@@ -0,0 +1,62 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_dataset_argument:
+
+********************
+``dataset`` argument
+********************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current implementation.
+
+All commands which operate on datasets have a ``dataset`` argument (``-d`` or
+``--dataset`` for the command line API) to identify a single dataset as the
+context of an operation.
+
+
+Impact on relative path resolution
+==================================
+
+With one exception, the nature of a provided ``dataset`` argument does **not**
+impact the interpretation of relative paths. Relative paths are always considered
+to be relative to the process working directory.
+
+The one exception to this rule is passing a ``Dataset`` object instance as
+``dataset`` argument value in the Python API. In this, and only this, case, a
+relative path is interpreted as relative to the root of the respective dataset.
```

Comment:
note: I always thought about it "in reverse", that CLI was a special case, not Python interface, but then realized that unbound invocations (i.e., like `datalad.api.clone`, without `dataset` being provided) also present in Python API.  So this formulation is probably the most succinct indeed to reflect the current state of the affairs.  But with "explicit better than implicit" (and the fact that `Dataset().<operation>` is passing `dataset` argument to operation is IMHO implicit here) I wonder if we better lay it out explicitly in addition (or instead) here:

```suggestion
relative path is interpreted as relative to the root of the respective dataset.

More specifically:

- relative paths in CLI are treated relative to :term:`CWD`
- relative paths in Python API will be treated 
  - to :term:`CWD` in invocations like ``datalad.api.<operation>(<arguments>)`` if ``<arguments>`` does not include ``dataset``
  - to context dataset root if 
    - aforementioned ``<operation>``invocation includes ``dataset`` argument
    - ``<operation>`` is performed as a `Dataset` object instance bound method, i.e. as ``Dataset(...).<operation>(<arguments>)``. **Note:** specification of ``dataset`` among ``<arguments>`` in such invocation will result in error.
```

Reply:
TBH I don't like that. I is probably saying the same thing, but even on third pass I am not confident that it is, or whether it add additional nuances.

Reply:
ok, let's not bother being explicit